src/com/alevel/hw6/Car.java

```diff
@@ -0,0 +1,91 @@
+package com.alevel.hw6;
+
+class Car {
+    private static final int MAX_FUEL = 100;
+
+    final private Manufacturer manufacturer;
+    private Engine engine;
+    final private Color color;
+    private int petrol;
+    private int mileage; //kilometers
+
+    public Car(Manufacturer manufacturer, String type, int power, Color color) {
+        this(manufacturer, new Engine(type, power), color);
+    }
+
+    public Car(Manufacturer manufacturer, Engine engine, Color color) {
+        this.manufacturer = manufacturer;
+        this.engine = engine;
+        this.color = color;
+        petrol = MAX_FUEL;
+        mileage = 0;
+    }
+
+    public void startEngine() {
+        if (petrol > 0) {
+            engine.startEngine();
+            System.out.printf("{%s} started the engine\n", manufacturer);
+        }
```

Comment:
A good addition is to check petrol, but if it is 0 it is better to write something